SwEnt-Group8/Street-work-app
# Feat/Improve progression screen with achievements linked to social interactions

## Overview
The objective of this pull request is to improve the progression screen with new achievements that are linked to social interactions between the users. As this is the main objective of our app, we wanted to reward the users who adds new friends with the bluetooth feature with dedicated achievements.

## Changes
- A new function `checkFriends` in the viewModel, in order to reward the user who added friends
- Achievements based on the number of friends
- Removed parts of progression screen (for instance the parks visited metric) that are not yet implemented, in order to aim for completeness in the screen


## How to Test

1. Add a friend
2. Go to the achievements part of the progression screen

You should have a new achievements for adding your first friend!


## Screenshot

<img width="241" alt="image" src="https://github.com/user-attachments/assets/7d951f4f-9208-49cf-b9f2-af1d9c3ed29c">
